addons/mail/static/src/messaging/entity/activity/activity.js

```diff
@@ -30,13 +31,105 @@ function ActivityFactory({ Entity }) {
             this.delete();
         }
 
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * @static
+         * @param {Object} data
+         * @return {Object}
+         */
+        static convertData(data) {
+            const data2 = {};
+            if ('activity_category' in data) {
+                data2.category = data.activity_category;
+            }
+            if ('can_write' in data) {
+                data2.canWrite = data.can_write;
+            }
+            if ('create_data' in data) {
+                data2.dateCreate = data.create_date;
+            }
+            if ('date_deadline' in data) {
+                data2.dateDeadline = data.date_deadline;
+            }
+            if ('force_next' in data) {
+                data2.force_next = data.force_next;
+            }
+            if ('icon' in data) {
+                data2.icon = data.icon;
+            }
+            if ('id' in data) {
+                data2.id = data.id;
+            }
+            if ('note' in data) {
+                data2.note = data.note;
+            }
+            if ('res_id' in data) {
+                data2.res_id = data.res_id;
+            }
+            if ('res_model' in data) {
+                data2.res_model = data.res_model;
+            }
+            if ('state' in data) {
+                data2.state = data.state;
+            }
+            if ('summary' in data) {
+                data2.summary = data.summary;
+            }
```

Comment:
I think this whole convert logic should be factorized, because it is always the same depending on the field type.

There should be a map between "server keys" and "fields", and instead of having to write the convert logic for every key manually, it should be written once per field type ideally, and then params are just iterated.

For attrs: always assignation if present, nothing if not present
For m2o: link if set, unlink if empty, nothing if not present
etc